src/test/java/com/stripe/model/TopupTest.java

```diff
@@ -0,0 +1,104 @@
+package com.stripe.model;
+
+import com.stripe.BaseStripeTest;
+import com.stripe.exception.StripeException;
+import com.stripe.net.APIResource;
+import com.stripe.net.LiveStripeResponseGetter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+public class TopupTest extends BaseStripeTest {
+    @Before
+    public void mockStripeResponseGetter() {
+        APIResource.setStripeResponseGetter(networkMock);
+    }
+
+    @After
+    public void unmockStripeResponseGetter() {
+        /* This needs to be done because tests aren't isolated in Java */
+        APIResource.setStripeResponseGetter(new LiveStripeResponseGetter());
+    }
+
+    @Test
+    public void testDeserialize() throws IOException {
+        String json = resource("topup.json");
+        Topup topup = APIResource.GSON.fromJson(json, Topup.class);
+        assertEquals("tu_HASH", topup.getId());
+        assertEquals("topup", topup.getObject());
+        assertEquals(1000, topup.getAmount().intValue());
```

Comment:
Hmm, is the `intValue` needed here?  This should automatically resolve right?

Reply:
Yes, we needed it here, since `Assert.assertEquals` has so many overloaded methods and there are only a few that works primitive types:
- http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(long, long)
- http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(double, double, double)
We can use http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(java.lang.Object, java.lang.Object) and do explicit boxing for expected value, e.g.:

```
assertEquals(Long.valueOf(1517619386), topup.getCreated());
```

Reply:
👍 Ah, tested this, that's sucks.

In any case, should probably use `longValue` for `getCreated` and `getExpectedAvailabilityDate` right?